indico/modules/events/registration/util.py

```diff
@@ -493,3 +493,10 @@ def generate_ticket(registration):
 
 def get_ticket_attachments(registration):
     return [('Ticket.pdf', generate_ticket(registration).getvalue())]
+
+
+def remove_regform_item_gaps(form_items, from_position):
+    """Update positions when deleting/disabling an item in order to prevent gaps"""
+    for item in form_items:
+        if item.position > from_position and item.is_enabled:
```

Comment:
Untested, but wouldn't something like this be safer? Unlike the current code it would ensure consistent positions on all items.

```python
positions = itertools.count(1)
disabled_positions = itertools.count(1)
for item in sorted(form_items, key=attrgetter('position')):
    pos = next(positions if item.is_enabled and not item.is_deleted else disabled_positions)
    item.position = pos
```

We could even go one step further and do this on the whole regform:

```python
section_positions = itertools.count(1)
disabled_section_positions = itertools.count(1000)
for section in sorted(regform.sections, key=attrgetter('position')):
    section_active = section.is_enabled and not section.is_deleted
    section.position = next(section_positions if section_active else disabled_section_positions)
    # ensure consistent field ordering
    positions = itertools.count(1)
    disabled_positions = itertools.count(1000)
    for child in section.children:
        child_active = child.is_enabled and not child.is_deleted
        child.position = next(positions if child_active else disabled_positions)
```

Reply:
Looks quite safe, if we really want to be cautions sounds like a good idea.